lib/fastlyctl/commands/logging/bigquery.rb

```diff
@@ -0,0 +1,125 @@
+module BigQuery
+
+    def BigQuery.parse_secret_key(file)
+        key = File.read(file)
+        if key[0..26] != "-----BEGIN PRIVATE KEY-----"
+            abort "Error, private key file should begin with -----BEGIN PRIVATE KEY-----"
+        end
+
+        return key 
+    end
+
+    def BigQuery.ensure_opts(required_opts,options) 
+        required_opts.each { |k| 
+            if options[k].nil? 
+                abort "Error, option #{k.to_s} is required for this action"
+            end
+        }
+    end
+
+    def BigQuery.print_configs(config)
+        max = {}
+        max["name"] = 0 
+        max["dataset"] = 0 
+        max["table"] = 0 
+        max["project_id"] = 0 
+        fields = ["name","dataset","table","project_id"]
+
+        config.each { |c| 
+            fields.each { |f| 
+                max[f] = c[f].length > max[f] ? c[f].length : max[f]
+           }
+        }
+        
+
+        puts
+        puts "Name".ljust(max["name"]) + " | " + "Dataset".ljust(max["dataset"]) + " | " + "Table".ljust(max["table"]) + " | " + "ProjectId".ljust(max["project_id"])
+        puts "-" * (max["name"] + max["dataset"] + max["table"] + max["project_id"])
+        config.each { |c| 
+            puts "%s | %s | %s | %s" % [c["name"].ljust(max["name"]), c["dataset"].ljust(max["dataset"]), c["table"].ljust(max["table"]), c["project_id"].ljust(max["project_id"])]
+        }
+        puts
+    end
+
+
+    def self.create(options)
+        puts "Creating bigquery log endpoint"
+        required_opts = ["name", "format_file", "user", "secret_key_file", "project_id", "dataset", "table" ]
+        
+        ensure_opts(required_opts,options)
+
+        parsed_key = parse_secret_key(options[:secret_key_file])
+
+        parsed_format = File.read(options[:format_file])
+
+        params = {}
+
+        id  = options[:service]
+        version = FastlyCTL::Fetcher.get_writable_version(id) unless options[:version]
+        version ||= options[:version]
+
+        params[:name] = options[:name] 
+        params[:format] = parsed_format 
+        params[:user] = options[:user]
+        params[:secret_key] = parsed_key
+        params[:project_id] = options[:project_id]
+        params[:dataset] = options[:dataset]
+        params[:table] = options[:table]
+
+        FastlyCTL::Fetcher.api_request(:post, "/service/#{id}/version/#{version}/logging/bigquery", params: params)
```

Comment:
I would change this to `body: params` just to be safe. `:params` is the query string and I feel like this might be too long for that in some cases. I could be wrong though.